### ISSUE TITLE
Auto update didn't work

### ISSUE THREAD

**Issue by vikramvi** [appium/appium-desktop#90]
<img width="654" alt="screen shot 2017-03-15 at 12 37 41 pm" src="https://cloud.githubusercontent.com/assets/308490/23946824/4ca26cc2-097c-11e7-8e49-e4cc2ba625b1.png">

Tapped on "Download Update Now" but nothing happened.

Mac 10.11.6 ( have current - 1 build installed )

**Comment by dpgraham** [comment]
This issue was fixed in 1.0.0-beta.4

Try clicking the 'Download Update Now' button and wait a few minutes. Eventually it should say 'Download Complete'. The download actually starts, it just doesn't show an indicator.

The only known issue left is that when you click to restart Appium it closes it without relaunching it and you have to wait up to a few minutes for it to install the new update before re-opening.

**Comment by vikramvi** [comment]
![screen shot 2017-03-15 at 5 30 05 pm](https://cloud.githubusercontent.com/assets/308490/23959286/12053634-09a5-11e7-9653-96e7a42a9ddb.png)

@dpgraham I waited for 5 mins and got this error.
It'll nice to show some fake loader that something is getting downloaded otherwise user will think something not correct.

can you please re-open this issue ?

**Comment by vikramvi** [comment]
@dpgraham Even 2nd time I got the same error. I've double checked that my company network is not behind some firewall / proxy.

**Comment by vikramvi** [comment]
@jlipps please re-open, I don't have permission to re-open closed issue.

**Comment by jlipps** [comment]
@vikramvi if you open up the web console (right click -> inspect) do you see anything in the logs for that window?

**Comment by vikramvi** [comment]
![screen shot 2017-03-15 at 5 59 19 pm](https://cloud.githubusercontent.com/assets/308490/23960787/3055a80e-09a9-11e7-80c4-4a45913806e0.png)

@jlipps  there is not network activity observed, also as I can't resize window it's hard to see what's going on there. I tried few times

**Comment by jlipps** [comment]
You'll need to hit CMD+R in that window to refresh it for network activity to be recorded

**Comment by vikramvi** [comment]
![screen shot 2017-03-15 at 6 05 01 pm](https://cloud.githubusercontent.com/assets/308490/23961036/f2eafe78-09a9-11e7-9c28-35f214978c70.png)

This is what I could see, is there a way to expand and get more verbose info ?

**Comment by jlipps** [comment]
No, not without rebuilding the app without window size restrictions. @dpgraham can you think of another way to look for logs?

**Comment by dpgraham** [comment]
If you reach the 'Could not download. An error has occurred' page it's because a server error occurred when attempting to download the latest update from GitHub releases.

The logs for this are stored at:

```
on Linux: ~/.config/appium-desktop/log.log
on OS X: ~/Library/Logs/appium-desktop/log.log
on Windows: %USERPROFILE%\AppData\Roaming\appium-desktop\log.log
```

In this version though (v1.0.0-beta.3) the autoUpdater info isn't being logged out so we may not get any helpful information (v1.0.0-beta.4 has these logs). Once we get v1.0.0-beta.5 out we'll be able to test this further (by downloading v1.0.0-beta.4 and having it update).

**Comment by dpgraham** [comment]
@vikramvi v1.0.0-beta.5 shows a download progress indicator. This was a bug.

**Comment by vikramvi** [comment]
@dpgraham  Thanks for detailed clarification. I will verify and close this issue after  v1.0.0-beta.5 is released.

**Comment by dpgraham** [comment]
Thanks, it will be released before end-of-week

And thanks for the help on this project. I appreciate it.

**Comment by GruLab** [comment]
![2017-03-16_09-42-04](https://cloud.githubusercontent.com/assets/8300079/23981737/e937cb10-0a2e-11e7-87ba-f9a01d3a4998.jpg)

Whenever i am starting Appium, it is getting autoupdated. After complete download "Click to restart appium" option is being shown, nothing updated after clicking it [i've waited for 10 minutes, still app not updated]

**Comment by dpgraham** [comment]
Hi @GruLab, this is a known issue on Windows that I submitted a patch for (https://github.com/appium/appium-desktop/commit/35562acd66c22a7b151faedf92d7d64b4a59490a), and it will be in the next beta (v1.0.0-beta.5). In the meantime, can you try uninstalling Appium and then installing the latest version (v1.0.0-beta.4) and let me know if the error happens again.

**Comment by GruLab** [comment]
Hi @dpgraham , Sure; i'll check it on "v1.0.0-beta.5". I've uninstalled the older one & installed the latest beta.4, no issue exists. ~Best Regards

**Comment by dpgraham** [comment]
Great, thanks @GruLab!

**Comment by GruLab** [comment]
Hi @dpgraham , issue still exists. 
![2017-03-26_23-41-57](https://cloud.githubusercontent.com/assets/8300079/24333900/259c6640-127e-11e7-9958-9cf9fd8045bc.png)
![2017-03-26_23-42-14](https://cloud.githubusercontent.com/assets/8300079/24333901/259d6d88-127e-11e7-880e-3511fe9f13c3.png)

**Comment by dpgraham** [comment]
Thanks for letting me know. This is proving to be quite difficult. I think what I'll do is replace the 'Could not download update. Try again later.' with a message and a link to the latest release.

**Comment by vikramvi** [comment]
@dpgraham Also I didn't find ~/Library/Logs/appium-desktop/log.log

There is no folder by name **appium-desktop** under Logs, please review

**Comment by GruLab** [comment]
@dpgraham You may do that in next release.

**Comment by dpgraham** [comment]
@vikramvi Have you searched your filesystem for a directory called 'appium-desktop'? They might be getting stashed somewhere else

@GruLab I've committed that fix, it'll be in our next version.

I'm going to close this issue because we're just going to go with the fallback of providing a download link whenever the update fails. Feel free to keep the discussion going on this ticket.

**Comment by vikramvi** [comment]
@dpgraham  I tried command `find / appium-desktop` from mac terminal and it didn't fetch any results.

**Comment by vikramvi** [comment]
@dpgraham @jlipps can we please re-open the ticket, as marking issue as closed gets ignore. I have posted few comments couple of days back but no update

Even with latest release, auto update is not working and as per release notes, there is no link which takes to download page.

**Comment by jlipps** [comment]
@vikramvi you won't see this new feature unless you have 1.6.4-beta.6 and there is a beta.7 update...

**Comment by GruLab** [comment]
@vikramvi Sorry for delayed response, i also didn't find any log on beta.5

@jlipps @dpgraham Waiting for beta.7, as i've replaced beta.5 with beta.6

**Comment by vikramvi** [comment]
@GruLab thanks for clarification.

@dpgraham can you please confirm if logging is working for any other users ?

@jlipps Let me uninstall existing version and download new one. Do you have plans to track auto update failures or just mention link to download latest version ?

**Comment by dpgraham** [comment]
We use https://www.npmjs.com/package/electron-log which is supposed to log out app data. The default location is `~/Library/Logs/<app name>/log.log` so not sure why you can't find it. Maybe it logged it somewhere else?

We'll keep track of auto-update failures. We've already put a fair bit of effort into trying to fix it without much success, I think it might be an issue with the library we're using. The fallback that we're going to use of just showing the link to the latest downloaded version, we've decided, will be good enough in the meantime.

**Comment by obstar** [comment]
still doesnt work. I try to update it on windows, it download it update ask me to close appium which I did, nothing happen for few minutes so I started appium desktop again and once more I get pop up about updating to new version (1.1.0-beta3)

**Comment by dpgraham** [comment]
Thanks for letting me know @obstar. I'll look into it again, this issue has been nagging us for a while.

**Comment by vikramvi** [comment]
@dpgraham  please review below

1. it will be nice to have some fake installation progress animation

2. can we capture auto update info in some local log file for easy debugging in case any issue occurs

**Comment by d-rai** [comment]
I am not able to update Appium 1.2.5 to 1.2.7.
<img width="656" alt="screen shot 2017-12-12 at 11 07 57 am" src="https://user-images.githubusercontent.com/24481771/33869498-f9a8611e-df2e-11e7-8c74-2f62d6a838df.png">

**Comment by Harsh-Mindtree** [comment]
To all those who are facing the issue "Not able to auto-update" after clicking on "Close appium" button.

There is a workaround for it. After the appium update gets downloaded and you see the pop-up, do not click on "Close appium" button rather then close appium directly using "x" icon at top of appium window and it will start auto-updating appium.

<img width="477" alt="appiumautoupdatesolution" src="https://user-images.githubusercontent.com/14075067/35276640-01994436-006a-11e8-9480-35651380f3f5.png">

I have been using this workaround for quite sometime and seeing it is successful everytime. Sharing it with everyone.

Hoping it helps:)

**Comment by dpgraham** [comment]
Thanks for sharing @Harsh-Mindtree ... that's interesting that it only works when you close it using the native exit.

**Comment by dpgraham** [comment]
Will publish to v1.4.1

**Comment by dpgraham** [comment]
Resolved with v1.4.1
